--- a/vllm/config/speculative.py	2026-09-23 19:36:24.946465964 -0400
+++ b/vllm/config/speculative.py	2026-09-01 22:56:29.365243272 -0400
@@ -50,6 +50,8 @@
     "exaone_moe_mtp",
     "exaone4_5_mtp",
     "qwen3_next_mtp",
+    "qwen4_exp",
+    "qwen4_exp_text",
     "qwen3_5_mtp",
     "longcat_flash_mtp",
     "bailing_hybrid_v3_mtp",
@@ -570,7 +572,7 @@
                 {"n_predict": n_predict, "architectures": ["NemotronHMTPModel"]}
             )
 
-        if hf_config.model_type == "qwen3_next":
+        if hf_config.model_type in ("qwen3_next", "qwen4_exp", "qwen4_exp_text"):
             hf_config.model_type = "qwen3_next_mtp"
         if hf_config.model_type == "qwen3_next_mtp":
             n_predict = getattr(hf_config, "num_nextn_predict_layers", None)
